install_config/downgrade.adoc

```diff
@@ -0,0 +1,439 @@
+= Downgrading OpenShift
+{product-author}
+{product-version}
+:icons: font
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap!:
+:description: Manual steps to revert OpenShift Enterprise to a previous version following an upgrade.
+:keywords: yum
+
+toc::[]
+
+== Overview
+
+Following an OpenShift Enterprise link:../install_config/upgrades.html[upgrade],
+it may be desirable in extreme cases to downgrade your cluster to a previous
+version. The following sections outline the required steps for each system in a
+cluster to perform such a downgrade, currently supported for the OpenShift
+Enterprise 3.1 to 3.0 downgrade path.
+
+[[downgrade-verifying-backups]]
+== Verifying Backups
+
+The Ansible playbook used during the
+link:../install_config/upgrades.html[upgrade process] should have created a
+backup of the *_master-config.yaml_* file and the etcd data directory. Ensure
+these exist on your masters and etcd members:
+
+====
+----
+/etc/openshift/master/master-config.yaml.<timestamp>
+/var/lib/openshift/etcd-backup-<timestamp>
+----
+====
+
+If you are using an external etcd cluster (versus the single embedded etcd), the
+backup is likely created on all etcd members, though only one is required for
+the recovery process.
+
+The RPM downgrade process in a later step should create *_.rpmsave_* backups of
+the following files, but it may be a good idea to keep a separate copy
+regardless:
+
+====
+----
+/etc/sysconfig/openshift-master
+/etc/etcd/etcd.conf <1>
+----
+<1> Only required if using external etcd.
+====
+
+[[downgrade-shutting-down-the-cluster]]
+== Shutting Down the Cluster
+
+On all masters, nodes, and etcd members (if using an external etcd cluster),
+ensure the relevant services are stopped:
+
+====
+----
+# systemctl stop atomic-openshift-master
+# systemctl stop atomic-openshift-node
+# systemctl stop etcd <1>
+----
+<1> Only required if using external etcd.
+====
+
+[[downgrade-removing-rpms]]
+== Removing RPMs
+
+On all masters, nodes, and etcd members (if using an external etcd cluster),
+remove the following packages:
+
+====
+----
+# yum remove atomic-openshift \
+    atomic-openshift-clients \
+    atomic-openshift-node \
+    atomic-openshift-master \
+    openvswitch \
+    atomic-openshift-sdn-ovs \
+    tuned-profiles-atomic-openshift-node
```

Comment:
Updated the package list here to remove `tuned-projects-atomic-openshift-node`, per https://bugzilla.redhat.com/show_bug.cgi?id=1297278.